osu.Game/Screens/Multi/Timeshift/TimeshiftMultiplayer.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Logging;
+using osu.Framework.Screens;
+using osu.Game.Screens.Multi.Lounge;
+using osu.Game.Screens.Multi.Match;
+
+namespace osu.Game.Screens.Multi.Timeshift
+{
+    public class TimeshiftMultiplayer : Multiplayer
+    {
+        protected override void UpdatePollingRate(bool isIdle)
+        {
+            var timeshiftManager = (TimeshiftRoomManager)RoomManager;
```

Comment:
This makes me think that maybe `Multiplayer` should be taken a step further and it should be a generic class, something of this sort, possibly?

```csharp
public abstract class Multiplayer<TRoomManager> : OsuScreen
	where TRoomManager : RoomManager
```

Would then allow to expose `RoomManager` as `TRoomManager` and avoid this cast, but unsure if too enterprise:tm:...

Reply:
Unfortunately `Multiplayer` is DI'd and used elsewhere in code. It's probably not the correct way of doing things but making `Multiplayer` generic feels like too much abstraction for very little gain.

Reply:
I think this is fine for now. Less generic has generally worked better for us until otherwise necessary (at the expense of an occasional explicit cast).